examples/decomposition/plot_faces_decomposition.py

```diff
@@ -120,57 +146,62 @@ def plot_gallery(title, images, n_col=n_col, n_row=n_row, cmap=plt.cm.gray):
 ]
 
 
-# #############################################################################
-# Plot a sample of the input data
-
-plot_gallery("First centered Olivetti faces", faces_centered[:n_components])
-
-# #############################################################################
-# Do the estimation and plot it
+# %%
+# Next, fit each estimator on all images and plot some results.
+# Each estimator extracts 6 components as vectors
+# :math:`h \in \mathbf{R}^{4096}`. We just displayed these vectors
+# in human-friendly visualisation as 64x64 pixel images.
+#
+# .. note::
+#
+#     The Eigenfaces estimator, via the :py:mod:`sklearn.decomposition.PCA`,
+#     also provides a scalar `noise_variance_` (the mean of pixelwise variance)
+#     that cannot be displayed as an image.
 
 for name, estimator, center in estimators:
-    print("Extracting the top %d %s..." % (n_components, name))
+    print("- Extracting the top %d '%s'..." % (n_components, name))
```

Comment:
I propose to break this for loop and instead create a cell where:

* we define the model
* and plot the gallery

It will have the benefit to remove the if/else for switching the behaviour depending of the estimator.